src/main/java/io/druid/data/input/Committer.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2003, 2010, Oracle and/or its affiliates. All rights reserved.
+ * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ */
+
+package io.druid.data.input;
+/**
+ * Committer includes a Runnable and a Jackson-serialized metadata object containing the offset
+ */
+public interface Committer extends Runnable
+{
+    public Object getMetadata();
```

Comment:
And include comments about the fact that it will need to be serialized and deserialized by Jackson, so it can be a complex type, but we recommend keeping it to List/Map/"Primitive JSON" types